worker/uniter/relation/relations_test.go

```diff
@@ -270,7 +272,7 @@ func (s *relationsSuite) TestNextOpNothing(c *gc.C) {
 		},
 	}
 	remoteState := remotestate.Snapshot{}
-	relationsResolver := relation.NewRelationsResolver(r)
+	_, relationsResolver := relation.NewRelationResolvers(r, nil)
```

Comment:
I realize it looks kinda ugly. The constructor signature is more of an indication to the caller that the two resolvers are wired to the same state tracker instance. 

My initial solution had two constructors but it felt it might not be apparent to the caller (note that if a different instance is passed things won't work as expected).

Open to suggestions though :-)

Reply:
Doh... why didn't I think of wrapping the two resolvers with an aux struct? I pushed https://github.com/juju/juju/pull/11341/commits/8526dd3f046a84a74d53a86fbaab8fffeba5416e which I think makes things much clearer.

Reply:
Ok, I will drop the last commit, rebase to remove the redundant charm.v6 bump and break this into two c-tors and we should be all set.